google/auth/_cloud_sdk.py

```diff
@@ -166,3 +103,27 @@ def load_authorized_user_credentials(info):
         token_uri=_GOOGLE_OAUTH2_TOKEN_ENDPOINT,
         client_id=info['client_id'],
         client_secret=info['client_secret'])
+
+
+def get_project_id():
+    """Gets the project ID from the Cloud SDK.
+
+    Returns:
+        Optional[str]: The project ID.
+    """
+
+    try:
+        output = subprocess.check_output([
+            'gcloud', 'config', 'config-helper', '--format', 'json'])
+    except (subprocess.CalledProcessError, OSError, IOError):
```

Comment:
because `CalledProcessError` will happen if gcloud returns anything other than 0 (for example, any sort of un-configured state), and depending on the OS / Python version it'll raise OSError or IOError if `gcloud` isn't installed.